### ISSUE TITLE
Schedule.xml/json: Add guid to room

### ISSUE THREAD

**Issue by saerdnaer** [pretalx/pretalx#1118]
When conference organisers change the room name, consuming scripts break. It would be really helpful if room also had an guid attribute, similar to lecture talks and other events.

At rC3 we added this room ids manually, while the source of the room guid was https://git.cccv.de/rc3/hub – but via the manual created mapping table at https://c3voc.de/wiki/events:rc3:schedule

I my opinion we should add an guid field to pretalx's room model, where crew users can fill in an existing guid – when it is left empty a random UUIDv4 is inserted automatically. It should be possible to change this field, as in some use cases the global uuid is not yet known when the room is created in pretalx.

The value of this new field should be rendered in the schedule.xml and schedule.json export. Example XML and JSONs are in https://github.com/frab/frab/issues/507 – the XSD already contains the necessary changes: https://github.com/voc/schedule/blob/master/validator/xsd/schedule.xml.xsd#L69

PS: We could also think about an additional room URI... e.g. in this example `https://pretalx.rc3.studio/rc3-channels-2020/rooms/8`:

```xml
<schedule>
  <version>1.7; rc3-channels 1.34; chaoszone 0.21; classics 1.0; oio 1.8; sendezentrum 0.14; wikipaka 0.93a-zielgerade2</version>
  <conference>
    <acronym>rc3</acronym>
    <title>Remote Chaos Experience</title>
    <start>2020-12-27</start>
    <end>2020-12-30</end>
    <days>4</days>
    <timeslot_duration>00:10</timeslot_duration>
    <time_zone_name>Europe/Berlin</time_zone_name>
  </conference>
  <day index="1" date="2020-12-27" start="2020-12-27T09:00:00+01:00" end="2020-12-28T04:00:00+01:00">
    <room name="rC1" guid="973ec154-a5b5-40ac-b4e8-b74137f647e8">...</room>
    <room name="rC2" guid="32b5ad05-ab7d-44eb-a6e6-7ca616eed34a">...</room>
    <room name="chaosstudio-hamburg" guid="008e96f4-1a64-46d3-af2e-7ae75fed1763">...</room>
    <room name="restrealitaet" guid="29f5ea95-2cb1-43bf-b3ea-83674646334d">...</room>
    <room name="r3s - Monheim/Rhein" guid="c4a577e2-52e7-4f6f-a5c0-e3822d64f84a" uri="https://pretalx.rc3.studio/rc3-channels-2020/rooms/8">
      <event id="65355" guid="ada0e6df-d5f3-5563-92e5-d7e52b07b2ab">
        <logo/>
        <date>2020-12-27T11:00:00+01:00</date>
        <start>11:00</start>
        <duration>00:14</duration>
        <room>r3s - Monheim/Rhein</room>
        <slug>rc3-65355-r3s_opening</slug>
        <url>
https://pretalx.rc3.studio/rc3-channels-2020/talk/SFGFYF/
        </url>
        <title>R3S Opening</title>
        <subtitle/>
        <track>MON r3s Rhein VHS</track>
        <type>Lightning Talk</type>
        <language>de</language>
        <abstract>Opening RheinRuhrStage R3S, Monheim / Rhein</abstract>
        <description/>
        <recording>
          <license/>
          <optout>false</optout>
        </recording>
        <persons>
          <person id="258">Glitzi</person>
        </persons>
        <links/>
        <attachments/>
      </event>
…
```

**Comment by rixx** [comment]
If somebody wants to build this feature, I'll be happy to review the PR.

**Comment by saerdnaer** [comment]
I think the PR is ready to be merged.

**Comment by rixx** [comment]
The feature will land before the next pretalx release. I'm currently holding off on the merge because deploying it will require a short maintenance window (since it involves an active data migration in production).

**Comment by rixx** [comment]
Closed via #1214
